code/modules/mob/living/simple_animal/bot/ed209bot.dm

```diff
@@ -192,8 +192,8 @@
 /mob/living/simple_animal/bot/ed209/attackby(obj/item/W, mob/user, params)
 	..()
 	if(istype(W, /obj/item/weldingtool) && user.a_intent != INTENT_HARM) // Any intent but harm will heal, so we shouldn't get angry.
-		return
-	if(!istype(W, /obj/item/screwdriver) && (!target)) // Added check for welding tool to fix #2432. Welding tool behavior is handled in superclass.
+		return // Added check for welding tool to fix #2432. Welding tool behavior is handled in superclass.
+	if(!isscrewdriver(W) && !locked && (!target)) //If the target is locked, they are recieving damage from the screwdriver
```

Comment:
Does `isscrewdriver()` check the tooltype or just whether it's a subtype of `/obj/item/screwdriver`? If so, is that deliberate?

Reply:
```
/proc/isscrewdriver(O)
	if(istype(O, /obj/item/screwdriver))
		return TRUE
	return FALSE
```
It checks if its a type of screwdriver, however items can be added to this proc if they are given screwdriver functionaliity. This was added back in #11700. I've tested it and it works for repairing bots with screwdrivers.

Reply:
why not make it a check for tool_behavior? I don't see why something that acts like a screwdriver wouldn't work here?

~~also `isscrewdriver` feels like it should be a define, not a proc, but oh well~~

Reply:
Yeah, those istool checks should probably be rewritten, but that would need to change all of them which is a bit out of scope.